### ISSUE TITLE
Planar parameters are not actually calculated.

### ISSUE THREAD

**Issue by sconlyshootery** [svip-lab/Indoor-SfMLearner#4]
https://github.com/svip-lab/Indoor-SfMLearner/blob/0d682b7ce292484e5e3e2161fc9fc07e2f5ca8d1/trainer_geo.py#L518
The planar parameters are always equal to zeros,  and I find the calculation of variables like points_channel, x_T_dot_x are not used in subsequent calucations.

**Comment by niujinshuchong** [comment]
@sconlyshootery 
In our experiments, the planar parameters are not zeros. It may be zeros at the beginning of the training if the predicted depth is zeros. I think you can use the pretrained model the check this.

x_T_dot_x is used. It was added to x_T_dot_X according to super pixel segmentation. Please not x_T_dot_x and x_T_dot_X are different. Also, point_channel is the channel of sum_points where sum_points refer to P_n^T Y_m in the paper. Then x_T_dot_X (P_n^T dot P_n in the paper) and sum_points was combined to calculate the plane parameters.

**Comment by sconlyshootery** [comment]
the sum_points is intialized as zeros, and never changes.
So xTx_inv_xT has to be zeros here.

**Comment by niujinshuchong** [comment]
@sconlyshootery 

scatter_add_ is an inplace operator. It will write the value to sum_points. If it's not the case in your environment, you may update PyTorch.

**Comment by sconlyshootery** [comment]
I see, thank you for your kind explanation. Another question, is it normal to get nan in planar depth, it is related to bad prediction of depth?

**Comment by sconlyshootery** [comment]
> I see, thank you for your kind explanation. Another question, is it normal to get nan in planar depth, it is related to bad prediction of depth?

The nan values are not from your codes, I use planar loss for my purpose, I don't use patchmatch, whether lack of it will lead to this phenomenon.

**Comment by niujinshuchong** [comment]
I think planar loss needs good depth initialization. You can add this loss when the predicted depth is reasonable.
